contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,168 @@
+
+Many kubernetes system components currently support these features.  Many more components that support these features are in development.
+The following is not meant to be an exhaustive list, but gives the current set of use cases for these metrics.
+
+Metrics requirements for "First Class Resource Isolation and Utilization Features", based on kubernetes component needs, are as follows:  
+
+ - Kubelet
+   - Node-level usage metrics for Filesystems, CPU, and Memory  
+   - Pod-level usage metrics for Filesystems and Memory  
+ - Metrics Server (outlined in [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md)), which exposes the [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md) to the following system components:
+   - Scheduler  
+     - Node-level usage metrics for Filesystems, CPU, and Memory  
+     - Pod-level usage metrics for Filesystems, CPU, and Memory  
+     - Container-level usage metrics for Filesystems, CPU, and Memory  
+   - Horizontal-Pod-Autoscaler  
+     - Node-level usage metrics for CPU and Memory  
+     - Pod-level usage metrics for CPU and Memory  
+   - Cluster Federation  
+     - Node-level usage metrics for Filesystems, CPU, and Memory  
+   - kubectl top and Kubernetes Dashboard  
+     - Node-level usage metrics for Filesystems, CPU, and Memory  
+     - Pod-level usage metrics for Filesystems, CPU, and Memory  
+     - Container-level usage metrics for Filesystems, CPU, and Memory  
+
+### Proposed Core Metrics:
+This section defines "usage metrics" for filesystems, CPU, and Memory.  
+As stated in Non-Goals, this proposal does not attempt to define the specific format by which these are exposed.  For convenience, it may be neccessary to include static information such as start time, node capacities for CPU, Memory, or filesystems, and more.
+
+```go
+// CpuUsage holds statistics about the amount of cpu time consumed  
+type CpuUsage struct {  
+  // The time at which these Metrics were updated.  
+  Timestamp metav1.Time  
+  // Cumulative CPU usage (sum of all cores) since object creation.  
+  CumulativeUsageNanoSeconds *uint64   
+}  
+
+// MemoryUsage holds statistics about the quantity of memory consumed  
+type MemoryUsage struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time  
+  // The amount of "working set" memory. This includes recently accessed memory,  
+  // dirty memory, and kernel memory.  
+  UsageBytes *uint64  
+}   
+
+// FilesystemUsage holds statistics about the quantity of local storage (e.g. disk) resources consumed  
+type FilesystemUsage struct {  
+  // The time at which these metrics were updated.  
+  Timestamp metav1.Time  
+  // This must uniquely identify the storage resource that is consumed.
```

Comment:
This will refer to the device/partition.  The namespace is the node.  I can update the comment,